lua/autorun/server/toggle_voice_init.lua

```diff
@@ -61,6 +61,9 @@ end )
 
 util.AddNetworkString( "proximity_voice_enabled_changed" )
 net.Receive( "proximity_voice_enabled_changed", function( _, ply )
-    local enabled = net.ReadBool()
-    playerConfig[ply] = enabled or nil
+    local enabled, transmitOnly = net.ReadBool(), nil
+    if enabled then
+        transmitOnly = net.ReadBool()
+    end
+    playerConfig[ply] = enabled and ( transmitOnly and 2 or 1 ) or nil
```

Comment:
This should be broken out into actual if statements and use constants to represent the states.
e.g.
```lua
local TRANSMIT_RECEIVE = 1
local TRANSMIT_ONLY = 2
```

```lua
if not enabled then 
    playrConfig[ply] = nil
elseif transmitOnly then
    playrConfig[ply] = TRANSMIT_ONLY
else
    playrConfig[ply] = TRANSMIT_RECEIVE
end
```

I think the nested ternary is just unnecessarily  confusing